From e4b2d12114d392d8b9c2e612c6cb1b7d338366cf Mon Sep 17 00:00:00 2001
From: John Thiltges <jthiltges2@unl.edu>
Date: Fri, 21 Jul 2017 12:05:07 -0500
Subject: [PATCH 1/3] Call globus_activate() even if HTTP support is not
 enabled

If HTTP support is not enabled, verification context creation fails. g_cert_dir is initialized in globus_activate(), which is only called from XrdHttpLcmaps. Add the globus_activate() call to XrdLcmaps as well.
---
 src/XrdLcmaps.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/XrdLcmaps.cc b/src/XrdLcmaps.cc
index 2dfc0bc..f99fad8 100644
--- a/src/XrdLcmaps.cc
+++ b/src/XrdLcmaps.cc
@@ -178,6 +178,8 @@ int XrdSecgsiAuthzInit(const char *cfg)
    int retval = XrdSecgsiAuthzConfig(cfg);
    if (retval) {return retval;}
 
+   if (!globus_activate()) {return -1;}
+
    // Done
    // 1 means 'OK and I want the certificate in PEM base64 format'
    return g_certificate_format;
-- 
2.7.5

